javaparser-core/src/main/java/com/github/javaparser/ast/body/ClassOrInterfaceDeclaration.java

```diff
@@ -364,6 +372,49 @@ public void ifClassOrInterfaceDeclaration(Consumer<ClassOrInterfaceDeclaration>
         action.accept(this);
     }
 
+    /**
+     * Returns whether this is a compact class (JEP 512 - implicit class from top-level members).
+     *
+     * @return true if compact, false otherwise
+     * @since 3.27.2
+     */
+    @Generated("com.github.javaparser.generator.core.node.PropertyGenerator")
+    public boolean isCompact() {
+        return isCompact;
+    }
+
+    /**
+     * Sets whether this is a compact class (JEP 512 - implicit class from top-level members).
+     *
+     * @param isCompact true if compact, false otherwise
+     * @return this declaration
+     * @since 3.27.2
+     */
+    @Generated("com.github.javaparser.generator.core.node.PropertyGenerator")
+    public ClassOrInterfaceDeclaration setCompact(final boolean isCompact) {
+        if (isCompact == this.isCompact) {
+            return this;
+        }
+        notifyPropertyChange(ObservableProperty.COMPACT, this.isCompact, isCompact);
+        this.isCompact = isCompact;
+        return this;
+    }
+
+    /**
+     * Returns the name of this class as a string.
+     * For compact classes (JEP 512), this returns an empty string to indicate the implicit/unnamed class.
+     *
+     * @return the class name, or empty string for compact classes
+     * @since 3.27.2
+     */
+    @Override
+    public String getNameAsString() {
+        if (isCompact) {
+            return "";
```

Comment:
I can see 2 issues here:
1. I'm not sure if the empty string is a good name to use. It makes the naming of nested classes awkward. I think that keeping a placeholder name would be better for consistent naming (but this is subjective so please share your thoughts on this @jlerbsc)
2. Overriding the name here is very brittle. `compactClass.getName().toString()` still returns `$CompactClass`, for example. I assume the pretty printer and various other mechanisms to get a string representation for the class would have the same issue.

Reply:
Does the JLS specify anything on this subject? Perhaps a test on a compact class containing an included class could improve our understanding of what needs to be done.